src/utils/parse.rs

```diff
@@ -0,0 +1,186 @@
+use crate::{model::prelude::*, prelude::*};
+
+// Questions:
+// - Should the error enums be #[non_exhaustive]?
+// - Should the MessageParseError implement source()? It may be too much boilerplate (maybe add thiserror dependency?)
+
+/// Parse a value from a string in context of a received message.
+///
+/// This trait is a superset of [`std::str::FromStr`] (the trait used in [`str::parse`]). The
+/// difference is that this trait supports serenity-specific Discord types like [`Member`],
+/// [`Role`] (soon), or [`Emoji`] (soon).
```

Comment:
If it does not support the mentioned items, they should not be listed here (see the *soon*), but items that it supports. Alternatively, generalise the sentence.